attribute_selector_block.py

```diff
@@ -0,0 +1,84 @@
+from enum import Enum
+
+from nio.signal.base import Signal
+from nio.block.base import Block
+from nio.properties import (VersionProperty, SelectProperty, ListProperty,
+                            Property, PropertyHolder)
+
+
+class Behavior(Enum):
+    BLACKLIST = False
+    WHITELIST = True
+
+
+class SpecItem(PropertyHolder):
+    item = Property(title='Attribute')
+
+
+class AttributeSelector(Block):
+    """
+    A block for whitelisting or blacklisting incoming signals and notifying
+    the rest.
+
+    Properties:
+    specify_behavior(select): select either whitelist or blacklist behavior
+    specify_attributes(list): list of incoming signal attributes to blacklist
+                              or whitelist
+    """
+
+    version = VersionProperty('1.0.0')
+    specify_behavior = SelectProperty(Behavior, title='Specify behavior',
+                                      default=Behavior.BLACKLIST)
+    specify_attributes = ListProperty(SpecItem,
```

Comment:
I think since the SpecItem only has one item, we can do `ListProperty(StringType...` to avoid having to create a whole new property holder and nest them.